src/common/interceptors/mask-sensitive-data.interceptor.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  Module,
+  NestInterceptor,
+} from "@nestjs/common";
+import { APP_INTERCEPTOR } from "@nestjs/core";
+import { instanceToPlain } from "class-transformer";
+import { isEmail } from "class-validator";
+import { map, Observable } from "rxjs";
+
+@Injectable()
+class MaskSensitiveDataInterceptor implements NestInterceptor {
+  private readonly keysToMask = ["orcidOfOwner"];
+
+  private maskSensitiveData<T>(data: T, ownEmail: string): T {
+    if (Array.isArray(data)) {
+      let allArrays = true;
+      const maskedData = data.map((item) => {
+        if (this.isToMaskEmail(item, ownEmail)) {
+          return this.maskValue();
+        }
+        allArrays = false;
+        return this.maskSensitiveData(item, ownEmail);
+      });
+      return ((allArrays && [...new Set(maskedData)]) || maskedData) as T;
+    }
+
+    if (!this.isPlainObject(data)) return data as T;
+    const result: Record<string, unknown> = {};
+    for (const [key, value] of Object.entries(data)) {
+      if (
+        this.keysToMask.includes(key) ||
```

Comment:
In this function, you are masking the ORCID only by targeting the field, but you are masking any email no matter where is found.
This does not cover the case where the ORCID is saved in the description or any other field. In such case, the ORCID will be returned to the user.
I think you should change the ORCID masking to follow the same principle as the email.

Reply:
that's a valid point, but I am not sure there's a way to "understand" if a format follows the ORCID one. Let me ask internally if we really need to mask this, maybe we don't and we limit this interceptor to hide emails for now